lib/matplotlib/tests/test_image.py

```diff
@@ -1233,6 +1233,24 @@ def test_imshow_quantitynd():
     fig.canvas.draw()
 
 
+@check_figures_equal(extensions=['png'])
+def test_norm_change(fig_test, fig_ref):
+    # LogNorm should not mask anything invalid permanently.
```

Comment:
Do you want to add in some invalids beforehand that should be kept throughout as well and not unmasked at any point?

Reply:
Sure, added.